src/context/context.service.ts

```diff
@@ -95,7 +95,11 @@ export class ContextService {
   }
 
   setHtmlBody(body: string): void {
```

Comment:
Worked for me but I noticed that it still wraps the loaded doc in `<html>` tags. A suggested modification below that would allow this function to be used by both the page-proxy and the new page api call so there's no need for two separate functions:

```suggestion
     setHtmlBody(
    body: string,
    loadAsDocument = false,
    divId = 'konviw-ontent',
  ): void {
    const $ = cheerio.load(body);
    // we wrap the body in a div with ID Content
    this.cheerioBody = cheerio.load(
      $('html').wrapInner(`<div id="${divId}">`).html(),
      null,
      loadAsDocument,
    );
  }
```
Explanation of the suggested new parameters (note that the default values are just proposals and there's no problem if you want to go with different defaults.

**`loadAsDocument`**: sets the cheerio `isDocument` property. If true it will load the page wrapped with `<html>`, if false it will load it as a fragment so it does not add `<html></html>` which is the desired behaviour for the page API (more info in the cheerio docs: https://cheerio.js.org/modules.html#load). I proposed the default as `false` as the method name gives the expectation that the body is loaded rather than an entire document.

**`divId`**: I think it would be useful to be able to customise this ID, especially as `Content` is quite generically used. For example if a site is embedding the page, there might be another div with `id=Content`, or if there is a need to embed two pages in one there would be duplicate div IDs. I proposed the default here as `konviw-content` to have a less generic ID, but I guess this might cause some issues in existing stylesheets so it's no issue if `Content` is used.